sql/core/src/main/scala/org/apache/spark/sql/SQLImplicits.scala

```diff
@@ -40,12 +40,15 @@ abstract class SQLImplicits extends LowPrioritySQLImplicits {
    */
   implicit class StringToColumn(val sc: StringContext) {
     def $(args: Any*): ColumnName = {
-      new ColumnName(sc.s(args: _*))
+      if (sc.parts.length == 1 && sc.parts.contains("*")) {
```

Comment:
what does this change fix?

Reply:
Yuo this is redundant. Removed it in the latest build